### PR TITLE
[mono][jit] Use an mrgctx for all gshared methods on WASM.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
Failures are unrelated.

File: src/mono/mono/utils/options-def.h
Comment (kotlarmilos) [inline src/mono/mono/utils/options-def.h:145]: Should it include `HOST_IOS` as well?

**Comment by vargaz** [comment]
The benchmarking suite can be run on wasm locally, not sure about ios.

**Comment by vargaz** [comment]
Also, it could be merged and later reverted if the perf impact is significant.

File: src/mono/mono/utils/options-def.h
Comment (vargaz) [reply src/mono/mono/utils/options-def.h:145]: Its better to start with one platform to see how it goes.

**Comment by kotlarmilos** [comment]
@vargaz is it ready to be merged after the Preview 4 snap? Is there anything else missing?

**Comment by vargaz** [comment]
It could be merged.

**Comment by kotlarmilos** [comment]
> It could be merged.

Good. I will approve it after the Preview 4 snap.

**Review by kotlarmilos — APPROVED** [review]
LGTM, thank you! Hopefully, we can reduce the size with reasonable speed tradeoff.

P.S. I would leave this to @lambdageek for final approval.

/cc: @SamMonoRT

**Comment by vargaz** [comment]
Will see what changes this will do to benchmarks.

**Pull request merged**
